axiom/test/test_userbase.py

```diff
@@ -637,52 +620,54 @@ def test_setPassword(self):
         L{LoginAccount.setPassword} allows for logging in with the new password
         and not the old.
         """
+        self.realm._txCryptContext, perform = getTestContext()
         account = self.realm.addAccount(
             self.localpart, self.domain, self.password)
         username = u'%s@%s' % (self.localpart, self.domain)
-        self.successResultOf(account.setPassword(u'blahblah'))
-        self.assertEquals(
-            self.successResultOf(
-                self._requestAvatarId(
-                    UsernamePassword(username, u'blahblah'))),
-            account.storeID)
-        self.failureResultOf(
-            self._requestAvatarId(
-                UsernamePassword(username, self.password)),
-            UnauthorizedLogin)
+        d = account.setPassword(u'blahblah')
+        perform()
+        self.successResultOf(d)
+        d = self._requestAvatarId(UsernamePassword(username, u'blahblah'))
+        perform()
+        self.assertEquals(self.successResultOf(d), account.storeID)
+        d = self._requestAvatarId(UsernamePassword(username, self.password))
```

Comment:
Might be worth adding another assertion that trying to authenticate with `account.passwordHash` will fail.